nixos/modules/services/networking/ctrld.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ctrld;
+  settingsFormat = pkgs.formats.toml {};
+in {
+
+  options.services.ctrld = {
+    enable = mkEnableOption "ctrld service";
+
+    settings = mkOption {
+      type = settingsFormat.type;
+
+      # https://github.com/Control-D-Inc/ctrld/blob/main/README.md#default-config
+      default = {
+        listener."listener.0" = {
+          ip = "";
+          port = 0;
+          restricted = false;
+        };
+
+        network."network.0" = {
+          cidrs = [ "0.0.0.0/0" ];
+          name = "Network 0";
+        };
+
+        service = {
+          log_level = "info";
+          log_path = "";
+        };
+
+        upstream = {
+          "upstream.0" = {
+            bootstrap_ip = "76.76.2.11";
+            endpoint = "https://freedns.controld.com/p1";
+            name = "Control D - Anti-Malware";
+            timeout = 5000;
+            type = "doh";
+          };
+
+          "upstream.1" = {
+            bootstrap_ip = "76.76.2.11";
+            endpoint = "p2.freedns.controld.com";
+            name = "Control D - No Ads";
+            timeout = 3000;
+            type = "doq";
+          };
+        };
+      };
+
+      description = ''
+        Configuration for ctrld, see
+        <link xlink:href="https://github.com/Control-D-Inc/ctrld/blob/release-branch-v${pkgs.ctrld.version}/docs/config.md">
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc."ctrld/ctrld.toml".source = settingsFormat.generate "ctrld.toml" cfg.settings;
+
+    systemd.services.ctrld = {
+      description = "ctrld proxy client";
+      startLimitIntervalSec = 5;
+      startLimitBurst = 10;
+      serviceConfig = {
+        ExecStart = "${getExe pkgs.ctrld} start --config ${cfg.configFile}";
+      };
+      after = [ "network.target" ];
+      before = [ "nss-lookup.target" ];
+      wants = [ "nss-lookup.target" ];
+      wantedBy = [ "multi-user.target" ];
+    };
+  };
+
+  meta.buildDocsInSandbox = false;
```

Comment:
Why is this needed?

Reply:
Because I'm using pkgs.ctrld.version in meta.description.